d3-dashboard/590.js

```diff
@@ -0,0 +1,176 @@
+  .attr('fill', 'transparent');
+
+const instagramLine = d3.line()
+  .x(d => xScale(d.year))
+  .y(d => yScale(d.followers.instagram));
+
+lineGraph.append('path')
+  .attr('d', instagramLine(data))
+  .attr('stroke', instagramColor)
+  .attr('stroke-width', 3)
+  .attr('fill', 'transparent');
+  
+lineGraph.selectAll('twitter-circles')
+  .data(data)
+  .enter()
+  .append('circle')
+  .attr('cx', d => xScale(d.year))
+  .attr('cy', d => yScale(d.followers.twitter))
+  .attr('r', 6)
+  .attr('fill', 'white')
+  .attr('stroke', twitterColor)
+  .style('cursor', 'pointer')
+
+lineGraph.selectAll('tumblr-circles')
+  .data(data)
+  .enter()
+  .append('circle')
+  .attr('cx', d => xScale(d.year))
+  .attr('cy', d => yScale(d.followers.tumblr))
+  .attr('r', 6)
+  .attr('fill', 'white')
+  .attr('stroke', tumblrColor)
+  .style('cursor', 'pointer')
+
+lineGraph.selectAll('instagram-circles')
+  .data(data)
+  .enter()
+  .append('circle')
+  .attr('cx', d => xScale(d.year))
+  .attr('cy', d => yScale(d.followers.instagram))
+  .attr('r', 6)
+  .attr('fill', 'white')
+  .attr('stroke', instagramColor)
+  .style('cursor', 'pointer')
+
+const rightDashboard = d3.select('.dashboard')
+  .append('div');
+
+const pieGraph = rightDashboard.append('svg')
+  .attr('width', 200)
+  .attr('height', 200)
+
+const pieArc = d3.arc()
+  .outerRadius(100)
+  .innerRadius(0);
+
+const pieColors = d3.scaleOrdinal()  
+  .domain(data[8].followers)
+  .range([twitterColor, tumblrColor, instagramColor]);
+
+const pie = d3.pie()
+  .value(d => d.value);
+  
+const pieGraphData = pieGraph.selectAll('pieSlices')
+  .data(pie(d3.entries(data[8].followers)))
+  .enter()
+  .append('g')
+  .attr('transform', 'translate(100, 100)');
+
+pieGraphData.append('path')
+  .attr('d', pieArc)
+  .attr('fill', d => pieColors(d.data.key))
+  .attr('stroke', 'white')
+  .attr('stroke-width', 2);
+
+pieGraphData.selectAll('pieSliceText')
+  .data(pie(d3.entries(data[8].followers)))
+  .enter()
+  .append('text')
+  .text(d => `${Math.round(d.data.value/d3.sum(d3.values(data[8].followers))*100)}%`)
+  .attr('transform', d => `translate(${pieArc.centroid(d)})`)
+  .style('text-anchor', 'middle')
+  .style('font', '10px verdana');
+
+const legend = rightDashboard.append('table')
+  .attr('width', 200)
+  .attr('height', 120)
+  .style('font', '12px verdana')
+
+const legendTitle = legend.append('thead')
+  .append('tr')
+  .append('th')
+  .text('2020 followers')
+  .attr('colspan', 3)
+
+const legendRows = legend.append('tbody')
+  .selectAll('tr')
+  .data(d3.entries(data[8].followers))
+  .enter()
+  .append('tr');
+
+legendRows.append('td')  
+  .text(d => d.key)
+  .attr('align', 'right');
+
+legendRows.append('td')
+  .attr('align', 'center')
+  .append('div')
+  .style('width', '16px')
+  .style('height', '16px')
+  .style('background-color', d => pieColors(d.key))
+
+legendRows.append('td')
+
+/*
+  Set the `text` to a `d` function that returns `d.value`.
```

Comment:
This is excellent. Maybe some of the earlier lines like lines 63 and 64 can rewritten to use this format rather than show the answer.